### PR TITLE
Fix a crash when an application error type has a non-numeric "status" property

### PR DESCRIPTION
**Pull request by lovasoa**
Hello @lukeed !

This is a follow-up on the sveltekit issue and https://github.com/lukeed/polka/commit/f95a5b4ad93315468cd16e62e69725a317e8ebc0

I hope this helps, and lets the discussions be less heated next time ;)

### CODE REVIEW

File: packages/polka/index.js [hunk 1]
```diff
@@ -3,7 +3,8 @@ import Router from 'trouter';
 import { parse } from '@polka/url';
 
 function onError(err, req, res) {
-	let code = (res.statusCode = err.status || 500);
+	const code = typeof err.status === 'number' ? err.status : 500;
+	res.statusCode = code;
```
Comment (lukeed) [inline packages/polka/index.js:7]: Avoid code style changes in a PR

```suggestion
	let code = (res.statusCode = +err.status || 500);
```

**Comment by lukeed** [comment]
Thanks for PR. 

PS It's not heated on my end – just trying to give you feedback as to how & why your reports come across as antagonistic. Appreciate the contribution

File: packages/polka/index.js (hunk 1 above)
Comment (lovasoa) [reply packages/polka/index.js]: Hmmm... This way, `err.status === true` will crash the server, for instance.

Maybe we could actually check `err.status in http.STATUS_CODES` ?

Comment (lukeed) [reply packages/polka/index.js]: I reverted it in the linked commit, sorry. The only rule (from Node pov) is that it has to be >= 100. You can send down a 700 status code if you want, which means that looking up `http.STATUS_CODES` isn't really an option unfortunately

**Pull request merged**